test/jdk/java/lang/Math/Clamp.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+   @bug 8301226
+   @summary Add clamp() methods to java.lang.Math
+ */
+
+
+public class Clamp {
+    public static void main(String[] args) {
+        int failures = 0;
+
+        failures += testIntClamp();
+        failures += testLongClamp();
+        failures += testDoubleClamp();
+        failures += testFloatClamp();
+
+        if (failures > 0) {
+            System.err.println("Testing clamp incurred " + failures + " failures.");
+            throw new RuntimeException();
+        }
+    }
+
+    private static int testIntClamp() {
+        int failures = 0;
+        failures += checkIntClamp(0, 1, 2, 1);
```

Comment:
Possible refactoring here: represent the test cases in, say, a 2D int array or an array/list of records and loop over them.

Reply:
Updated. Is it better this way?